### PR TITLE
[PLAT-382] Include META-INF/NOTICE and META-INF/LICENSE in JARs

### LINKED ISSUE #18567: NOTICE files and Licenses in the distributions are broken
IMDG distribution includes incorrect NOTICE and Licenses files.

Regarding NOTICE, this file should be included in distribution:
- https://github.com/hazelcast/hazelcast/blob/master/NOTICE

Regarding Licenses it should be discussed with legal department what needs to be done and what would be acceptable from the legal point of view.

### PR DESCRIPTION
**Pull request by kwart**
Include proper `NOTICE` and `LICENSE`  in our JARs.
Resolves #18567.

### CODE REVIEW

File: hazelcast/LICENSE [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (frant-hartm) [inline hazelcast/LICENSE:190]: This should be current year and hazelcast?

File: hazelcast-sql/pom.xml [hunk 2]
```diff
@@ -143,8 +141,6 @@
                                 </relocation>
                             </relocations>
                             <transformers>
-                                <transformer
```
Comment (frant-hartm) [inline hazelcast-sql/pom.xml:146]: Why is this removed?

**Review by frant-hartm — CHANGES_REQUESTED** [review]

File: hazelcast/LICENSE (hunk 1 above)
Comment (kwart) [reply hazelcast/LICENSE:190]: I don't think so. This is just an appendix saying how to apply it in files:
```
      To apply the Apache License to your work, attach the following
      boilerplate notice, with the fields enclosed by brackets "[]"
      ...
```

Comment (frant-hartm) [reply hazelcast/LICENSE:190]: Ah right.

File: hazelcast-sql/pom.xml (hunk 2 above)
Comment (kwart) [reply hazelcast-sql/pom.xml:146]: This is actually a pretty good question and I will need to dig more into it. There seems to be some problem with the transformer.
Ref: https://github.com/apache/maven-shade-plugin/blob/maven-shade-plugin-3.2.4/src/main/java/org/apache/maven/plugins/shade/resource/ApacheLicenseResourceTransformer.java

When we use the transformer, then no license file is included in the resulting JAR. When we don't use it, then 2 files are added - `LICENSE` and `LICENSE.txt`.
Unfortunatelly, it's not the only difference in the JARs!
The root in the JAR **with** the transformer contains 2 directories and 1 file:
```
com
META-INF
com.hazelcast.org.codehaus.commons.compiler.properties
```

The root in the JAR **without** the transformer contains 4 directories and 2 files:
```
com
io
META-INF
nonapi
com.hazelcast.org.codehaus.commons.compiler.properties
LICENSE-ClassGraph.txt
```

**Review by olukas — COMMENTED** [review]
LGTM but I rather suggest to ask @Holmistr for review. He can discuss it with legal department.

**Comment by kwart** [comment]
run-lab-run

File: hazelcast-sql/pom.xml (hunk 2 above)
Comment (kwart) [reply hazelcast-sql/pom.xml:278]: I used our own transformer instead - the `HazelcastLicenseResourceTransformer`.

**Review by olukas — APPROVED** [review]

File: hazelcast-build-utils/src/main/resources/hazelcast-thirdparty-template.ftl [hunk 3]
```diff
@@ -301,6 +286,130 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 -----
 
+Creative Commons Legal Code
```
Comment (frant-hartm) [inline hazelcast-build-utils/src/main/resources/hazelcast-thirdparty-template.ftl:289]: This file should contain all permitted licenses (allowed in the root pom and in the distribution pom).

**Review by frant-hartm — APPROVED** [review]

**Review by frant-hartm — CHANGES_REQUESTED** [review]
Please add the permitted licences to the thirdparty template.

**Review by frant-hartm — APPROVED** [review]

**Comment by kwart** [comment]
Thanks for the reviews!
